src/CoreWCF.Http/tests/SimpleWSHTTPTest.cs

```diff
@@ -163,6 +169,105 @@ internal void WSHttpRequestReplyWithTransportMessageCertificateEchoString()
             }
         }
 
+#if NETCOREAPP3_1_OR_GREATER // Windows Auth on Kestrel not supported on NetFx
+        [Fact, Description("transport-security-with-windows-authentication-kestrel")]
+        [Trait("Category", "NetCoreOnly")]
+        internal void WSHttpRequestImpersonateWithKestrel()
+        {
+            string testString = new string('a', 3000);
+            IWebHost host = ServiceHelper.CreateHttpsWebHostBuilder<WSHttpTransportWithImpersonation>(_output).Build();
+            using (host)
+            {
+                host.Start();
+                System.ServiceModel.WSHttpBinding wsHttpBinding = ClientHelper.GetBufferedModeWSHttpBinding(System.ServiceModel.SecurityMode.Transport);
+                var factory = new System.ServiceModel.ChannelFactory<ClientContract.IEchoService>(wsHttpBinding,
+                    new System.ServiceModel.EndpointAddress(new Uri("https://localhost:8443/WSHttpWcfService/basichttp.svc")));
+                factory.Credentials.Windows.AllowedImpersonationLevel = TokenImpersonationLevel.Impersonation;
+                factory.Credentials.ServiceCertificate.SslCertificateAuthentication = new System.ServiceModel.Security.X509ServiceCertificateAuthentication
+                {
+                    CertificateValidationMode = System.ServiceModel.Security.X509CertificateValidationMode.None
+                };
+                ClientContract.IEchoService channel = factory.CreateChannel();
+                ((IChannel)channel).Open();
+                string result = channel.EchoForImpersonation(testString);
+                Assert.Equal(testString, result);
+                ((IChannel)channel).Close();
+            }
+        }
+#endif
+
+        [Fact, Description("transport-security-with-windows-authentication-httpsys")]
+        internal void WSHttpRequestImpersonateWithHttpSys()
+        {
```

Comment:
This test is failing on Linux due to Linux not supporting HttpSys. You need to add a windows only trait. You do that by adding the following attribute:
```c#
[Trait("Category", "WindowsOnly")]
```

Reply:
Added the WindowsOnly attribute